src/switch.zig

```diff
@@ -0,0 +1,379 @@
+const lsquic = @cImport({
+    @cInclude("lsquic.h");
+    @cInclude("lsquic_types.h");
+    @cInclude("lsxpack_header.h");
+});
+const std = @import("std");
+const libp2p = @import("root.zig");
+const quic = libp2p.transport.quic;
+const protocols = libp2p.protocols;
+const Allocator = std.mem.Allocator;
+
+/// The Switch struct is the main entry point for managing connections and protocol handlers.
+/// It acts as a central hub for handling incoming and outgoing connections,
+/// as well as managing protocol handlers for different protocols.
+/// It supports multiple transports, but currently only one transport is supported at a time.
+/// The Switch is responsible for creating and managing connections, listeners, and protocol handlers.
+/// It also provides methods for dialing peers, listening for incoming connections,
+/// and handling protocol messages.
+pub const Switch = struct {
```

Comment:
It's unfortunate that libp2p calls this thing a [switch](https://docs.libp2p.io/concepts/appendix/glossary/#switch)...
It's typically best to avoid a language keyword, mainly for the practical annoyance of having to escape it everywhere. For example, it seems to break github's zig syntax highlighter.

Reply:
Yes, I found this problem, let me change to `swarm` which is an alias in libp2p.